include/slb/functional.hpp

```diff
@@ -252,6 +252,121 @@ detail::not_fn_result<FD> not_fn(F&& f) {
 }
 #endif
 
+// [func.bind], bind
+
+#if SLB_INTEGRAL_CONSTANT == 2 // C++14
+using std::is_bind_expression;
+using std::is_placeholder;
+#else
+template <typename T>
+struct is_bind_expression
+    : slb::bool_constant<std::is_bind_expression<T>::value> {};
+
+template <typename T>
+struct is_placeholder
+    : slb::integral_constant<int, std::is_placeholder<T>::value> {};
+#endif
+
+// We only enable the C++17 implementation under C++2a here to account for
+// P0704: "Fixing const-qualified pointers to members" and LWG2210: "INVOKE-ing
+// a pointer to member with a `reference_wrapper` as the object expression".
+
+#if __cpp_lib_invoke /* C++17 */ && __cplusplus > 201703L /* C++2a */
+using std::bind;
+#else
+namespace detail {
+struct bind_tag {
+  explicit bind_tag() = default;
+};
+
+template <typename FD>
+class bound {
+  FD fd;
+
+public:
+  template <typename F>
+  bound(bind_tag, F&& f) : fd(std::forward<F>(f)) {}
+
+  template <typename... Args>
+  /*constexpr*/ auto operator()(Args&&... args) noexcept(noexcept(
```

Comment:
The `constexpr` here and below are commented out because it's unspecified as to whether these should/can be `constexpr`? Are we trying to figure out what to do here? If we can't add it, shouldn't we just remove it?

Reply:
These could be `constexpr` but we are not allowed to make them so. This is a leftover from me testing that they were.